tests/OptimizedPathMappingRepositoryTest.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository\Tests;
+
+use Puli\Repository\Api\EditableRepository;
+use Puli\Repository\Api\Resource\Resource;
+use Puli\Repository\OptimizedPathMappingRepository;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+use Puli\Repository\Resource\DirectoryResource;
+use Puli\Repository\Resource\FileResource;
+use Puli\Repository\Tests\Resource\TestFilesystemDirectory;
+use Puli\Repository\Tests\Resource\TestFilesystemFile;
+use Webmozart\KeyValueStore\ArrayStore;
+
+/**
+ * @since  1.0
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class OptimizedPathMappingRepositoryTest extends AbstractEditableRepositoryTest
+{
+    /**
+     * @var ArrayStore
+     */
+    protected $store;
+
+    /**
+     * @var OptimizedPathMappingRepository
+     */
+    protected $repo;
+
+    /**
+     * Counter to avoid collisions during tests on files
+     *
+     * @var int
+     */
+    protected static $createdFiles = 0;
+
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->store = new ArrayStore();
+        $this->repo = new OptimizedPathMappingRepository($this->store);
+    }
+
+
+    protected function createPrefilledRepository(Resource $root)
+    {
+        $repo = new OptimizedPathMappingRepository(new ArrayStore());
+        $repo->add('/', $root);
+
+        return $repo;
+    }
+
+    protected function createWriteRepository()
+    {
+        return new OptimizedPathMappingRepository(new ArrayStore());
+    }
+
+    protected function createReadRepository(EditableRepository $writeRepo)
+    {
+        return $writeRepo;
+    }
+
+    protected function createFile($path = null, $body = TestFilesystemFile::BODY)
+    {
+        $filesystemPath = __DIR__ . '/Fixtures/path_mapping/file' . self::$createdFiles;
```

Comment:
You should work in the system's temporary directory, not in the fixtures directory, otherwise we might commit files created by the test by accident. See FilesystemRepositoryTest::setUp() and tearDown() for inspiration.